src/deprecates.jl

```diff
@@ -40,4 +40,19 @@ for fun in [:pdf, :logpdf,
     end
 end
 
-@deprecate pdf(d::DiscreteUnivariateDistribution) pdf.(Ref(d), support(d))
+@deprecate pdf(d::UnivariateDistribution{<:ContiguousSupport}) pdf.(Ref(d), support(d))
+
+# No longer proposing use of many const aliases
+const ValueSupport = Support{Float64}
+const Discrete = ContiguousSupport{Int}
+const Continuous = ContinuousSupport{Float64}
+
+const DiscreteDistribution{F<:VariateForm}   = Distribution{F,Discrete}
+const ContinuousDistribution{F<:VariateForm} = Distribution{F,Continuous}
+
+const DiscreteUnivariateDistribution     = Distribution{Univariate,    Discrete}
+const ContinuousUnivariateDistribution   = Distribution{Univariate,    Continuous}
+const DiscreteMultivariateDistribution   = Distribution{Multivariate,  Discrete}
+const ContinuousMultivariateDistribution = Distribution{Multivariate,  Continuous}
+const DiscreteMatrixDistribution         = Distribution{Matrixvariate, Discrete}
+const ContinuousMatrixDistribution       = Distribution{Matrixvariate, Continuous}
```

Comment:
Can we use `Base.@deprecate_binding` for these?

Reply:
good idea thanks, it preserves compatibility but signals what is changing

Reply:
Cool. I had no idea that existed - I see it isn't exported in fact, so I don't feel too bad. Anyway, I'll definitely do that.

Reply:
I notice that it doesn't handle parametric types, so I can't do it for all of the deprecated types:

```julia
const DiscreteDistribution{F<:VariateForm}   = Distribution{F,Discrete}
const ContinuousDistribution{F<:VariateForm} = Distribution{F,Continuous}
```

but I've covered most of them now. However, testing them is another matter:

```julia
julia> @test_deprecated DiscreteMatrixDistribution
WARNING: Distributions.DiscreteMatrixDistribution is deprecated, use Distribution{Matrixvariate, Discrete} instead.
  likely near REPL[6]:1
Log Test Failed at REPL[6]:1
  Expression: $(Expr(:escape, :DiscreteMatrixDistribution))
  Log Pattern: (:warn, r"deprecated"i, Ignored(), :depwarn) match_mode = :any
  Captured Logs:

ERROR: There was an error during testing
```

Is the message returned by `@test_deprecated` not a deprecation warning?